spec/features/post_show_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.feature 'Post Show Page' do
+  scenario 'displays post details and comment details' do
+    # Assuming you have some users, posts, and comments in your test database
+    user = User.create(name: 'Ikoote rasuli', photo: 'user_photo.jpg', bio: 'I am good', posts_counter: 5)
+    post = Post.create(user:, title: 'Post Title', text: 'Post text')
+    Comment.create(user:, post:, text: 'Comment 1')
+    Comment.create(user:, post:, text: 'Comment 2')
+
+    visit user_post_path(user, post)
+
+    expect(page).to have_content('POST DETAILS')
+    expect(page).to have_content('Post Title')
+    expect(page).to have_content('Number of comments:')
+    expect(page).to have_content('Number of likes:')
+    expect(page).to have_content('Post text')
+
+    expect(page).to have_content('Ikoote rasuli')
+    expect(page).to have_content('Comment 1')
+    expect(page).to have_content('Comment 2')
```

Comment:
- [ OPTIONAL ] I highly recommend wrapping each test in the `it` block since each test should have an explanation of what  is the test about so for example a `Test that asserts you can see the user's profile picture.`  then you can put its own test below it ✅✨🚀💯 but I am assigning it as optional because it is not required bu it is highly recommended ✨✅🙏🏻